src/libopensc/card-myeid.c

```diff
@@ -47,6 +47,10 @@
 #define MYEID_STATE_ACTIVATED	0x07
 
 #define MYEID_INFINEON_CHIP_ATR		0x04
+#define MYEID_CARD_NAME_MAX_LEN		100
+
+static const char *myeid_card_name = "MyEID";
+static char card_name_buf[MYEID_CARD_NAME_MAX_LEN];
```

Comment:
See myeid_init. card->name is set to point to this buffer, and it cannot be unallocated when returning, so a local variable won't do. I changed this to a global earlier, because when it was implemented as a local, card->name pointed to garbage later. Do you have another solution to this?